scheds/rust/scx_layered/src/layer_core_growth.rs

```diff
@@ -246,7 +248,19 @@ impl<'a> LayerCoreOrderGenerator<'a> {
             .nodes
             .clone()
             .into_iter()
-            .map(|(_, x)| x.all_cpus.into_iter().map(|(_, x)| x.id).collect())
+            .map(|(_, x)| {
+                x.all_cores
+                    .clone()
+                    .into_iter()
+                    .flat_map(|(_, x)| {
+                        if self.topo.all_cores.len() * 2 == self.topo.all_cpus.len() {
```

Comment:
Use [`self.topo.smt_enabled`](https://github.com/sched-ext/scx/blob/main/rust/scx_utils/src/topology.rs#L188) for a more robust check.

Reply:
That would fall over in (I think) a non-graceful manner on intel P/E core consumer chips (i.e. putting CPUs that don't exist in the vec).